controllers/ingressnodefirewall_controller.go

```diff
@@ -420,3 +458,25 @@ func mergeFirewallProtocolRules(a, b []infv1alpha1.IngressNodeFirewallProtocolRu
 	}
 	return a, nil
 }
+
+func stringInArray(val string, array []string) bool {
+	for i := range array {
+		if array[i] == val {
+			return true
+		}
+	}
+	return false
+}
+
+func removeString(s string, slice []string) (result []string, found bool) {
+	if len(slice) != 0 {
+		for _, item := range slice {
+			if item == s {
+				found = true
+				continue
+			}
+			result = append(result, item)
+		}
+	}
+	return
+}
```

Comment:
this is more effective, just FYI
https://github.com/golang/go/wiki/SliceTricks#filter-in-place